### ISSUE TITLE
pred可运行代码，仅供参考

### ISSUE THREAD

**Issue by pityk3369** [Lapis-Hong/atec-nlp#6]
`
# predict模型预测
# from demo.train import FLAGS

FLAGS.model_dir = '/home/lab-215/yangkang/githubRepository/atec-nlp/model'

FLAGS.max_document_length = 30  # 根据上面定义的flags来取值，"max_document_length", 30,
lines = ['好好学习天天向上\t我要起早贪黑的学习 1']

input_file = "/home/lab-215/yangkang/githubRepository/atec-nlp/data/pred.csv"
output_file = "/home/lab-215/yangkang/githubRepository/atec-nlp/data/testPred.csv"

print("\nPredicting...\n")
graph = tf.Graph()
with graph.as_default():  # with tf.Graph().as_default() as g:
    sess = tf.Session()
    with sess.as_default():
        

        meta_file = os.path.abspath(os.path.join(
            FLAGS.model_dir, 'checkpoints/model-2600.meta'))# Load the saved meta graph and restore variables

        new_saver = tf.train.import_meta_graph(meta_file)
        new_saver.restore(sess, tf.train.latest_checkpoint(
            os.path.join(FLAGS.model_dir, 'checkpoints')))

        input_x1 = graph.get_tensor_by_name("input_x1:0")  # Get the placeholders from the graph by name
        input_x2 = graph.get_tensor_by_name("input_x2:0")
        dropout_keep_prob = graph.get_tensor_by_name("dropout_keep_prob:0")  # 修改=注释
        
        y_pred = graph.get_tensor_by_name("y_pred:0") # Tensors we want to evaluate

        e = graph.get_tensor_by_name("cosine:0")

        # Generate batches for one epoch
        dataset = Dataset(data_file=input_file, is_training=False)
        data = dataset.process_data(
            data_file=input_file, sequence_length=FLAGS.max_document_length)
        batches = dataset.batch_iter(data, FLAGS.batch_size, 1, shuffle=False)
        with open(output_file, 'w') as fo:
            lineno = 1
            for batch in batches:
                #                 x1_batch, x2_batch, _, _ = zip(*batch) 修改
                x1_batch, x2_batch = zip(*batch)
                
                y_pred_ = sess.run(
                    [y_pred], {input_x1: x1_batch, input_x2: x2_batch, dropout_keep_prob: 1.0})
                for pred in y_pred_[0]:
                    fo.write('{}\t{}\n'.format(lineno, pred))
                    lineno += 1
`

**Comment by pityk3369** [comment]
dropout_keep_prob = graph.get_tensor_by_name("dropout_keep_prob:0")  # 修改=注释
y_pred = graph.get_tensor_by_name("y_pred:0") # Tensors we want to evaluate

这俩个变量作者在train阶段name定义失误了，大家记得修改哈
